packages/twenty-front/src/modules/object-record/record-table/components/RecordTableBody.tsx

```diff
@@ -12,9 +15,53 @@ type RecordTableBodyProps = {
 export const RecordTableBody = ({
   objectNameSingular,
 }: RecordTableBodyProps) => {
-  const { tableRowIdsState } = useRecordTableStates();
+  const {
+    tableRowIdsState,
+    isRecordTableInitialLoadingState,
+    visibleTableColumnsSelector,
+  } = useRecordTableStates();
 
   const tableRowIds = useRecoilValue(tableRowIdsState);
+  const visibleTableColumns = useRecoilValue(visibleTableColumnsSelector());
+
+  const isRecordTableInitialLoading = useRecoilValue(
+    isRecordTableInitialLoadingState,
+  );
+  const skeletonColumnsWithSmallWidth = ['Domain Name', 'ICP', 'ARR', 'X'];
+
+  if (isRecordTableInitialLoading && tableRowIds.length === 0) {
```

Comment:
Could we extract all this logic in another component RecordTableBodyLoading ? Just keep the predicate here